### PR TITLE
(Improve order flow) Rework fill in sample ids

### PR DESCRIPTION
**Pull request by islean**
## Description

Closes https://github.com/Clinical-Genomics/improve-order-flow/issues/95. 

### Added

-

### Changed

- _fill_in_sample_ids operates on the Sample object introduced in the improve-order-flow project

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/orders/submitters/order_submitter.py [hunk 1]
```diff
@@ -34,14 +33,13 @@ def store_order(self, order_in: OrderIn):
         pass
 
     @staticmethod
-    def _fill_in_sample_ids(samples: list[dict], lims_map: dict, id_key: str = "internal_id"):
+    def _fill_in_sample_ids(samples: list[Sample], lims_map: dict):
```
Comment (islean) [inline cg/services/orders/submitters/order_submitter.py:36]: The `Sample` is only used for new samples in the order, meaning that all samples in the input will miss their lims id. So we should be able to remove the check if the sample already has one set

File: cg/services/order_validation_service/models/sample.py [hunk 2]
```diff
@@ -8,6 +8,7 @@ class Sample(BaseModel):
     comment: str | None = None
     container: ContainerEnum
     container_name: str | None = None
+    generated_lims_id: str | None = None  # Will be populated by LIMS and stored as the internal id
```
Comment (islean) [inline cg/services/order_validation_service/models/sample.py:11]: Technically a tech-savvy user with curl knowledge can set their own internal ids with this change. Is that something we want to fix now or is that for the future?

**Review by diitaz93 — APPROVED** [review]
👍

File: cg/services/order_validation_service/models/sample.py (hunk 2 above)
Comment (diitaz93) [reply cg/services/order_validation_service/models/sample.py]: fixed with the private attribute

**Pull request merged**